### DISCUSSION TITLE
Needing compiled ImageMagickObject Setup !

### DISCUSSION THREAD

**Discussion by AlexLaforge** [ImageMagick/ImageMagick#4967] [Help] [answered]
Hi,

I'm using **ImageMagickObject COM+ on Windows**.
This interface sadly was recently (2021) _deprecated_ by ImageMagick !
cf. https://github.com/ImageMagick/ImageMagick/blob/main/ChangeLog.md#710-20---2022-01-22

I am desperately looking to **download the latest version supporting it, 7.1.0-19 - 2021-12-22.**
I am looking for the following compiled archives, but cannot find them on their website anymore

- ImageMagick-7.1.0-19-Q8-x64-dll.exe
- ImageMagick-7.1.0-19-Q16-x64-dll.exe

Could someone with a good soul give them to me ?

Many thanks... you would be a life saver !

**Comment by snibgo** [comment]
I have ImageMagick-7.1.0-19-Q16-HDRI-x64-dll.exe, it that is any use.

**Reply by AlexLaforge** [reply]
Hi @snibgo , that will probably be a version with too much memory usage for me, as I prefer Q8 no HDRI.
But, while waiting for **THE miracle answer** from another member, yes, I accept your kind proposal!
Would you send me a link in PM?
Many thanks @snibgo :)

**Comment by snibgo** [comment]
I have uploaded it to snibgo.com/im/ImageMagick-7.1.0-19-Q16-HDRI-x64-dll.exe (38 MB)

I will leave it there for a day or so.

**Reply by AlexLaforge** [reply]
Downloaded! Thanks ^^

**Reply by OlesSirota** [reply]
Hi, Alan.
I found your link on the forum with the latest version of the «ImageMagickObject.dll».
But the link «snibgo.com/im/ImageMagick-7.1.0-19-Q16-HDRI-x64-dll.exe» contains a file of zero length.
Can you help me?
Yours, Serge Sirota.

**Reply by snibgo** [reply]
I suppose I overwrote it. Sorry.

**Reply by OlesSirota** [reply]
File lost? Maybe you still have it in your local archives?

**Comment by dlemstra** [comment, marked as answer]
You can find other files here: https://github.com/ImageMagick/ImageMagick/actions/runs/1613048022 in the installers zip file.

**Reply by AlexLaforge** [reply]
Oh @dlemstra ! I was not aware of the fact that these "Artifacts" are still archived here on GitHub! You **ARE** a real life saver.
How long are these files retained?
One question however : **Where did you get the hyperlink you sent me**?
In the "Release" page, I can only find that 7.1.0-19 is leading to https://github.com/ImageMagick/ImageMagick/actions/runs/1627888256, which is not the same as your hyperlink...

**Reply by AlexLaforge** [reply]
Additionally do you know where I can find the discussions and explanations that motivated the ImageMagickObject interface support dropping?
I'm probably not the only one feeling alone and lost in the forest now... :(

**Reply by dlemstra** [reply]
I have no idea how long those files will be retained. I found that link by going through our GitHub actions builds: https://github.com/ImageMagick/ImageMagick/actions and searched for a build around a release date.

There were no discussions about ImageMagickObject. We stopped supporting it because we no longer wanted to support this hacky module and coder. And we will not support it again.

**Reply by AlexLaforge** [reply]
@dlemstra Yes, I did the same, by digging around the action builds date/time. Thanks for this tip.

About ImageMagickObject, I understand you won't re-introduce it, OK. Please forgive the question that will follow : Instead of using an instance the blind ("**WScript.Shell**") from VBScript or WSH or Classic ASP, is there a chance I can call the ImageMagick .NET interface from it? (eg. like this : https://www.codeproject.com/Articles/79314/How-to-Call-a-NET-DLL-from-a-VBScript-2)

**Reply by dlemstra** [reply]
I don't think that is possible because it requires the code to be `ComVisible` and the Magick.NET library does not do this. And with Classic ASP being end of life in 2025 it is probably time to switch to something else.

**Reply by AlexLaforge** [reply]
Thank you @dlemstra for this info. I know that `ComVisible` should be included while compiling Magick.NET. Did not knew if it was.

Concerning Classic ASP, yes it is ancient, but still very alive an used by a lot of major actors in many industries, making it very difficult for Microsoft to drop the ASP.DLL.

As Windows 2022 was released in 2021, September, Classic ASP will be running at least until 10 years after this release date, so leading us to **2031** : cf. https://docs.microsoft.com/en-us/troubleshoot/developer/webapps/iis/general/asp-support-windows
